docs/design/deviceshifu/design-deviceShifu-lwm2m.md

```diff
@@ -0,0 +1,216 @@
+- normal mode: just like the other deviceShifu, when call the instruction, deviceShifu will send Read or Write Request to deviceShifu and return response
+- [observe mode](https://guidelines.openmobilealliance.org/object-support/#observe-and-notify-multiple-resources): this mode will set the min and max notify time, then device will notify data when data changed or timeout. deviceShifu will record data, and return the data store in the memory cache when call the instruction. and this kind of mode also support read and write operation.
+
+deviceShifu will host a LwM2M server and listen on UDP 3683(LwM2M default port) and HTTP server(deviceShifu) on 8080.
+
+If the object is in observe mode, the LwM2M server must also handle notify requests from the device and update the data in the memory cache.
+
+If deviceShifu receives an instruction before the device is registered, it will return an error message.
+
+```mermaid
+sequenceDiagram
+    participant d as device
+    participant ds as deviceShifu
+    participant ap as Application
+    participant bs as BootStrap Server
+    
+    alt bootstrap is enable
+    d ->> bs: Get Server Info
+    bs ->> d: Reply Server Info
+    end
+
+    d ->> ds: Register
+    ds ->> d: Created
+
+    alt observe mode
+    ds ->> d: Observe[Write the Notification Attributes Settings]
+    d ->> ds: Success
+    ds ->> d: Observe
+    loop Device Data Change or Timeout
+        d ->> ds: Notify
+        ds ->> d: Update Device Data
+    end
+    else normal mode
+        alt read
+            ap ->> ds: Read Object
+            ds ->> d: Read
+            d ->> ds: Data
+            ds ->> ap: Data
+        else write
+            ap ->> ds: Write Data
+            ds ->> d: Write Object Data
+            d ->> ds: Changed
+            ds ->> ap: Changed
+        end
+    end
+```
+
+### Protocol Specification
+
+Define data structures and types in Go for configuring and managing LwM2M communication:
+
+```go
+type LwM2MSetting struct {
+	EndpointName string `json:"endpointName,omitempty"`
+  	// +kubebuilder:default="None"
+	SecurityMode *SecurityMode `json:"securityMode,omitempty"`
+	DTLSMode     *DTLSMode     `json:"dtlsMode,omitempty"`
+
+	CipherSuites []CiperSuite `json:"cipherSuites,omitempty"`
+	PSKIdentity  *string      `json:"pskIdentity,omitempty"`
+	PSKKey       *string      `json:"pskKey,omitempty"`
+}
+
+type DTLSMode string
+
+const (
+    DTLSModePSK DTLSMode = "PSK"
+    // ...
+)
+
+type CiperSuite string
+
+const (
+	CiperSuite_TLS_ECDHE_ECDSA_WITH_AES_128_CCM   CiperSuite = "TLS_ECDHE_ECDSA_WITH_AES_128_CCM"
+    // ...
+)
+```
+
+```go
+type LwM2MType string
+
+type Properties struct {
+    ObjectId            string      `json:"ObjectId"` // required example /3303/0
+    DataType            LwM2MType   `json:"DataType"` // optional float, int, bool, string, default string
+    EnableObserve       bool        `json:"EnableObserve,omitempty"` // optional enable observe mode default false
+}
+```
```

Comment:
are these for `protocolPropertyList` in ConfigMap? if so add a section title to this file